src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,1296 @@
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation7()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_CommentsAfter()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y) //Comments");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$ //Comments");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_CommentsAfter2()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y) //Comments");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$ //Comments");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation8_SemicolonAlreadyExists()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y);");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);;$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_MultiLine()
+        {
+            var code = CreateTestWithMethodCall(@"
+var test = ClassC.MethodM(
+    x$$, 
+    y)");
+
+            var expected = CreateTestWithMethodCall(@"
+var test = ClassC.MethodM(
+    x, 
+    y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_MultiLine2()
+        {
+            var code = CreateTestWithMethodCall(@"
+var test = ClassC.MethodM(
+    x$$, 
+    y");
+
+            var expected = CreateTestWithMethodCall(@"
+var test = ClassC.MethodM(
+    x, 
+    y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_MultiLine3_knownfailure()
+        {
+            // Not yet handling scenario where closing paren is on the next line
```

Comment:
:memo: this case doesn't seem compelling to change at this time. Rather than calling it a known failure, I would document it as one possible outcome pending user feedback.